WebAppManifestClient.js

```diff
@@ -70,39 +73,32 @@ export class WebAppManifestClient {
 
   async getManifestWithIcon({
     colorScheme = 'light', defaultIcon, origin, size
-  }) {
-    const {defaultHeaders: headers, agent} = this;
+  } = {}) {
     let manifest;
     let icon = {};
     try {
       manifest = await this.getManifest({origin});
     } catch(err) {
-      const favicon = await getFavicon({headers, agent, origin});
-      favicon ? icon.src = favicon : icon.src = defaultIcon;
-      if(!icon.src) {
-        icon = undefined;
-      }
+      defaultIcon ? icon.src = defaultIcon : icon.src =
+        `${origin}/favicon.ico`;
       return {manifest, icon};
```

Comment:
This is simpler, right?

```suggestion
      icon.src = defaultIcon || `${origin}/favicon.ico`;
      return {manifest, icon};
```